src/QueryPlanVisualizer/Helpers/EntityFrameworkDatabaseHelper.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Data.Common;
+using System.Data.Entity.Infrastructure.Interception;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ExecutionPlanVisualizer.Helpers
+{
+    internal class EntityFrameworkDatabaseHelper : DatabaseHelper
+    {
+        public override string GetSqlServerQueryExecutionPlan<T>(IQueryable<T> queryable)
+        {
+            var interceptor = new CommandCapturingInterceptor();
+            try
+            {
+
+                DbInterception.Add(interceptor);
+
+                try
+                {
+                    var result = queryable.Provider.Execute(queryable.Expression);
+                }
+                catch (Exception ex)
+                {
+                    if (ex is CommandCapturedException || ex.InnerException is CommandCapturedException)
+                    {
+                        // ignore
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+
+                if (interceptor.Command == null)
+                {
+                    throw new InvalidOperationException("DbInterception failed to capture DbCommand.");
+                }
+
+                Connection = interceptor.Command.Connection;
+
+                Connection.Open();
+
+                try
+                {
+                    using (var command = Connection.CreateCommand())
+                    {
+                        command.CommandText = "SET STATISTICS XML ON";
+                        command.ExecuteNonQuery();
+                    }
+
```

Comment:
Yep, you're right. All the derived class really needs to do is provide the DbCommand for the base class to run. It's a lot cleaner now.